src/SFML/Window/GlResource.cpp

```diff
@@ -38,6 +38,13 @@ GlResource::GlResource()
 }
 
 
+////////////////////////////////////////////////////////////
+GlResource::GlResource(GlResource&&) noexcept
+{
+    priv::GlContext::initResource();
```

Comment:
Nice, this is a solution to get `resourceCount` updated correctly.
However, a mutex lock is involved, which might throw according to [cppreference](https://en.cppreference.com/w/cpp/thread/recursive_mutex/lock), so I am wondering if it is suitable for a `noexcept` function.

Reply:
I am still new to source code, but wouldn't be more cleaner to have a singleton class to manage `resourceCount` and other global variable related? if it was like that we had a ref to the singleton class in `GlResource` that would be moved and we did not needed to increase `resourceCount` to 2 and moved object doesn't have valid ref so it couldn't do anything when goes out of scope.

Reply:
@kimci86 I removed `noexcept` but static analyzer is unhappy.
what do you suggest ?

Reply:
Maybe there is a smarter way but one solution I can think of is to keep `noexcept` on move operations and add a boolean variable in `GlResource` telling if `priv::GlContext::cleanupResource` needs to be called in `GlResource` destructor. This makes possible to implement move operations with simple boolean variable assignments.

Reply:
Actually it will be cleaner to wait for #2503 to get merged as it adds move semantics for `GlResource`.